activerecord/lib/arel/collectors/composite.rb

```diff
@@ -3,7 +3,7 @@
 module Arel # :nodoc: all
   module Collectors
     class Composite
-      attr_accessor :preparable
+      attr_accessor :preparable, :retryable
```

Comment:
Technically this (and `preparable`) should propagate to the composed `left` and `right` collectors when assigned 🤔

Reply:
Done (`retryable` only, don't want to mix changes to `preparable` into this PR 😅 )